backend/homework/4/src/main/java/org/example/question2/TicketReservation.java

```diff
@@ -0,0 +1,62 @@
+package org.example.question2;
+
+import java.util.*;
+
+public class TicketReservation {
+    private static final int CONFIRMEDLIST_LIMIT = 10;
+    private static final int WAITINGLIST_LIMIT = 3;
+
+    private final List<Passenger> confirmedList = new ArrayList<>();
+    private final Deque<Passenger> waitingList = new ArrayDeque<>();
+
+    public List<Passenger> getConfirmedList(){
+        return confirmedList;
+    }
+
+    public Deque<Passenger> getWaitingList() {
+        return waitingList;
+    }
+
+    public boolean cancel(String confirmationNumber) {
+        Iterator<Passenger> confirmedIterator = confirmedList.iterator();
+        boolean checkRemove = removePassenger(confirmedIterator, confirmationNumber);
+
+        if(checkRemove){
+            if(!waitingList.isEmpty()){
+                Passenger nextPassenger = waitingList.poll();
+                confirmedList.add(nextPassenger);
+            }
+            return true;
+        }
+        else{
+            Iterator<Passenger> waitingIterator = waitingList.iterator();
+            return removePassenger(waitingIterator, confirmationNumber);
+        }
+    }
+
+    public boolean bookFlight(String firstName, String lastName, int age, String gender, String travelClass, String confirmationNumber) {
+        Passenger passenger = new Passenger(firstName, lastName, age, gender, travelClass, confirmationNumber);
+        if(confirmedList.size() < CONFIRMEDLIST_LIMIT){
+            confirmedList.add(passenger);
+            return true;
+        }
+        else if(waitingList.size() < WAITINGLIST_LIMIT){
+            waitingList.add(passenger);
+            return true;
+        }
+        return false;
```

Comment:
Lets not return false here, rather throw some exception and handle it.

Reply:
It was told in the question itself to return false